vhdbuilder/packer/configure-windows-vhd.ps1

```diff
@@ -426,13 +426,117 @@ function Update-WindowsFeatures {
+        New-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides"
+    }
+
+    $currentValue=(Get-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides" -Name $Name -ErrorAction Ignore)
+    if (![string]::IsNullOrEmpty($currentValue)) {
+        Write-Log "The current value of $Name in FeatureManagement\Overrides is $currentValue"
+    }
+    Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides" -Name $Name -Value $Value -Type $Type
+}
+
+function Enable-WindowsFixInHnsState {
+    Param(
+      [Parameter(Mandatory = $true)][string]
+      $Name,
+      [Parameter(Mandatory = $false)][string]
+      $Value = "1",
+      [Parameter(Mandatory = $false)][string]
+      $Type = "DWORD"
+    )
+
+    $currentValue=(Get-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\hns\State" -Name $Name -ErrorAction Ignore)
+    if (![string]::IsNullOrEmpty($currentValue)) {
+        Write-Log "The current value of $Name in hns\State is $currentValue"
+    }
+    Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\hns\State" -Name $Name -Value $Value -Type $Type
+}
+
+function Enable-WindowsFixInVfpExtParameters {
+    Param(
+      [Parameter(Mandatory = $true)][string]
+      $Name,
+      [Parameter(Mandatory = $false)][string]
+      $Value = "1",
+      [Parameter(Mandatory = $false)][string]
+      $Type = "DWORD"
+    )
+
+    $regPath=(Get-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Services\VfpExt" -ErrorAction Ignore)
+    if (!$regPath) {
+        Write-Log "Creating HKLM:\SYSTEM\CurrentControlSet\Services\VfpExt"
+        New-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Services\VfpExt"
+    }
+
+    $regPath=(Get-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Services\VfpExt\Parameters" -ErrorAction Ignore)
+    if (!$regPath) {
+        Write-Log "Creating HKLM:\SYSTEM\CurrentControlSet\Services\VfpExt\Parameters"
+        New-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Services\VfpExt\Parameters"
+    }
+
+    $currentValue=(Get-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\VfpExt\Parameters" -Name $Name -ErrorAction Ignore)
+    if (![string]::IsNullOrEmpty($currentValue)) {
+        Write-Log "The current value of $Name in VfpExt\Parameters is $currentValue"
+    }
+    Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\VfpExt\Parameters" -Name $Name $Value -Type $Type
+}
+
+function Enable-WindowsFixInPath {
+    Param(
+      [Parameter(Mandatory = $true)][string]
+      $Path,
+      [Parameter(Mandatory = $true)][string]
+      $Name,
+      [Parameter(Mandatory = $false)][string]
+      $Value = "1",
+      [Parameter(Mandatory = $false)][string]
+      $Type = "DWORD"
+    )
+    $regPath=(Get-Item -Path $Path -ErrorAction Ignore)
+    if (!$regPath) {
+        Write-Log "Creating $Path"
+        New-Item -Path $Path
```

Comment:
Question: Do you need to create the upper-level path if the upper-level path does not exist? Like function `Enable-WindowsFixInFeatureManagement` and `Enable-WindowsFixInVfpExtParameters`.

Reply:
I suggest having a generic function to create registry key path like below:
- merge similar code
- return early if exist

```
function Create-RegistryKeyPath {
    Param(
        [Parameter(Mandatory = $true)][string] $Path
    )

    $regPath=(Get-Item -Path $Path -ErrorAction Ignore)
    if (!$regPath) {
        # Create the upper level registry key path if it does not exist
        $upperLevelPath = [IO.Path]::GetDirectoryName($Path)
        Create-RegistryKeyPath -Path $upperLevelPath

        # Create the current registry key path
        Write-Log "Creating $Path"
        New-Item -Path $Path
    }
}
```

Reply:
I prefer to handle these specific cases one by one instead of using this generic function to handle it as the function `Enable-WindowsFixInVfpExtParameters`.
And currently new registry keys are in FeatureManagement or hns\State. 
Let us do this improvement if it is needed in future.